simulator/infra/types.h

```diff
@@ -39,25 +39,15 @@ using float64 = double;
 // Use native GCC type if available, as Boost <= 1.60 + GCC 7 generate a bug
 #if defined(__GNUC__) && defined(__SIZEOF_INT128__)
 
+#define USE_GNUC_INT128
+
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wpedantic"
 
```

Comment:
`#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wpedantic"`

What are these for there btw?

Reply:
GCC’s uint128 is not a standard C++, so compiler emits a warning. These pragmas silence warnings around that piece of code.